tools/mason/MasonNew.chpl

```diff
@@ -135,12 +137,13 @@ proc beginInteractiveSession() throws {
         var currChapelVersion = getChapelVersionStr();
         write("Chapel version (" + currChapelVersion + "): ");
         IO.stdout.flush();
-        readf("%s", chapelVersion);
-        if chapelVersion == currChapelVersion then gotCorrectChapelVersion = true;
+        IO.stdin.readline(chapelVersion);
+        chapelVersion = chapelVersion.strip();
         if chapelVersion == "" then chapelVersion = currChapelVersion;
         if chapelVersion != currChapelVersion then 
           throw new owned MasonError("This version of Chapel is deprecated. Please use" +
               " latest version " + currChapelVersion);
```

Comment:
I've added a check here to make sure that an user is using the latest version of Chapel. Should there be an option for an user to use any other previous version here ? Also, could this error message be improved ?

Reply:
I think so. We should probably assume the user knows what they are doing if they override the default. 

We should probably print a warning if the user selects a version range that is not compatible with their current version, e.g. `0.1.3` or `1.19..1.20`

Reply:
@ben-albrecht Are users allowed to use a version range for `chplVersion` ? 
Also, how would compatibility of versions be checked ? Can you elaborate this ?

Reply:
We can replace the `MasonError` message here with a `Note` which would advice users to use latest version of Chapel

Reply:
> Are users allowed to use a version range for chplVersion ?
> Also, how would compatibility of versions be checked ? Can you elaborate this ?

Yes, it uses range syntax, where no range implies a "this version or above", or `x...`. Examples: `1.19.0` == `1.19.0..`, `1.20..1.21`, or `1.22.0..1.22.0` (only works with 1.22)

Reply:
> Also, how would compatibility of versions be checked ? 

I think we just need to check that the current chapel version is within the specified range.

Reply:
@ben-albrecht I found that this check is already done by `mason update`. I feel this feature is not very necessary for interactive, since, we are already providing default chapel version hint while taking input. If an user makes a mistake and uses an incompatible version, I believe it should be `mason update` and `mason build` which should display them the error message rather than interactive.

Reply:
Would it be difficult to call the same validation function already used in `mason update` in the interactive session?

I agree that it's somewhat redundant, but it could also be a nice way to quickly catch and correct a user error, rather than letting them complete the interactive setup and learn of their mistake afterwards.

For example:

```sh
> mason new --interactive
This is an interactive session to walk you through creating a library project using Mason.
The following queries covers the common items required to create the project. 
It also tries to guess sensible defaults.

Press <somebutton> to quit. 


name (default displayed here) : BensPkg
version (default displayed) :  0.1.0
chplVersion (default displayed) : 1.20.0..1.22.0
Warning: Your current Chapel version (1.23.0) is not compatible with this chplVersion.
Are you sure you want to use: 1.20.0..1.22.0 ? [Y/n]

...
```

It's a bit of a UI micro-optimization, so I'm fine either way.